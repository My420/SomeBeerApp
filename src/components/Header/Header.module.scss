@import '../../sass/variables.scss';

.header {
  display: flex;
  align-items: center;

  width: $PC-container-width;
  min-height: 60px;

  margin: 0 auto;
  padding: 0 20px;
}

.inner {
  display: flex;
  justify-content: center;
  width: $PC-max-width;
}

.button {
  position: absolute;
  right: 20px;

  padding: 0px;

  background-color: transparent;
  border: none;

  cursor: pointer;
}

.button:focus {
  outline: 2px solid $menu-hover-color;
}

@media (max-width: $PC-max-width) {
  .header {
    width: $MPC-container-width;
  }

  .inner {
    display: flex;
    width: $MPC-container-width;
  }
}

@media (max-width: $MPC-max-width) {
  .header {
    position: fixed;

    width: 100%;
    height: 60px;

    padding: 0;
    background-color: $menu-bar-color;
  }

  .inner {
    display: block;
    position: relative;
    width: 100%;
    height: 60px;
  }
}
