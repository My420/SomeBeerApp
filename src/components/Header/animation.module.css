.appear {
}
.appearActive {
}
.appearDone {
}
.enter {
  transform: translate(-100%, 60px);
}
.enterActive {
  transform: translate(0, 60px);
  transition: all 300ms;
}
.enterDone {
}
.exit {
  transform: translate(0, 60px);
}
.exitActive {
  transform: translate(-100%, 60px);
  transition: all 300ms;
}
.exitDone {
}
