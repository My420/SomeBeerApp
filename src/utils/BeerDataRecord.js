import { Record } from 'immutable';

const BeerDataRecord = Record(
  {
    id: null,
    name: null,
    tagline: null,
    first_brewed: null,
    description: null,
    image_url: null,
    abv: null,
    ibu: null,
    target_fg: null,
    target_og: null,
    ebc: null,
    srm: null,
    ph: null,
    attenuation_level: null,
    volume: {},
    boil_volume: {},
    method: {},
    ingredients: {},
    food_pairing: [],
    brewers_tips: null,
    contributed_by: null
  },
  'BeerDataRecord'
);

export default BeerDataRecord;
