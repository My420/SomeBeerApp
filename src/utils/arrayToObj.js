/* eslint no-param-reassign: "error" */

const arrayToObj = arr => {
  return arr.reduce((prevValue, element) => {
    prevValue[element.id] = element;
    return prevValue;
  }, {});
};

export default arrayToObj;
