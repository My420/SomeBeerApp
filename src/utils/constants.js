export const SHOP_NAME = 'Some Beer Shop';

export const LINK = {
  HOME: '/',
  CATALOG: '/catalog',
  ROULETTE: '/roulette',
  ABOUT: '/about',
  FAVORITE: '/favorite',
  CART: '/cart'
};

export const NAVIGATION_LINKS = [
  [LINK.HOME, 'Home'],
  [LINK.CATALOG, 'Catalog'],
  [LINK.ROULETTE, 'Roulette!'],
  [LINK.FAVORITE, 'Favorite'],
  [LINK.ABOUT, 'About'],
  [LINK.CART, 'Cart']
];

export const BROWSER_WIDTH_RATIO = {
  LPC: 4,
  MPC: 3,
  TABLET_9: 2,
  TABLET_7: 1,
  MOBILE: 0
};

export const BROWSER_MIN_WIDTH = {
  LPC: 1200,
  MPC: 993,
  TABLET_9: 768,
  TABLET_7: 480,
  MOBILE: 320
};

export const SOCIAL_LINK = {
  GITHUB: 'https://github.com/My420',
  EMAIL: 'mailto:klinovitsky.aleksey@gmail.com',
  SKYPE: 'skype:live:klinovitsky.aleksey?chat'
};

export const AUTHOR = 'Klinovitsky Aleksey';

export const POPULAR_BEER__GET_REQUEST = 'POPULAR_BEER__GET_REQUEST';
export const POPULAR_BEER__GET_SUCCESS = 'POPULAR_BEER__GET_SUCCESS';
export const POPULAR_BEER__GET_FAILURE = 'POPULAR_BEER__GET_FAILURE';

export const API_HOST = 'https://api.punkapi.com/v2/';

export const POPULAR_BEER_AMOUNT = 8;

export const LOAD_ERROR_MESSAGE =
  'Could not download data, please refresh the page';

export const ERROR_TITLE = 'Oops! Something went wrong.';
export const ERROR_ACTION = 'Please refresh page or try again later.';
