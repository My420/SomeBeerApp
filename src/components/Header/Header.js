import React from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import styles from './Header.module.scss';
import animation from './animation.module.css';
import Navigation from '../Navigation/Navigation';
import Button from '../Button/Button';
import Container from '../Container/Container';
import { NAVIGATION_LINKS } from '../../utils/constants';
import IconCrossCss from '../IconCrossCss/IconCrossCss';
import IconMenuCss from '../IconMenuCss/IconMenuCss';

const Header = ({ isOpen, toggleState, isButtonShown }) => {
  // eslint-disable-next-line no-console
  console.log('render ===== Header');
  return (
    <header className={styles.header}>
      <Container className={styles.inner}>
        <TransitionGroup component={null}>
          {isOpen ? (
            <CSSTransition classNames={{ ...animation }} timeout={300}>
              <div>
                <Navigation links={NAVIGATION_LINKS} />
              </div>
            </CSSTransition>
          ) : null}
        </TransitionGroup>
      </Container>
      {isButtonShown ? (
        <Button onClick={toggleState} className={styles.button}>
          {isOpen ? <IconCrossCss /> : <IconMenuCss />}
        </Button>
      ) : null}
    </header>
  );
};

Header.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggleState: PropTypes.func.isRequired,
  isButtonShown: PropTypes.bool.isRequired
};

export default Header;
