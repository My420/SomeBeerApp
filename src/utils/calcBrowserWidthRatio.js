import { BROWSER_MIN_WIDTH, BROWSER_WIDTH_RATIO } from './constants';

const calcBrowserWidthRatio = width => {
  if (width > BROWSER_MIN_WIDTH.LPC) {
    return BROWSER_WIDTH_RATIO.LPC;
  }
  if (width > BROWSER_MIN_WIDTH.MPC) {
    return BROWSER_WIDTH_RATIO.MPC;
  }
  if (width > BROWSER_MIN_WIDTH.TABLET) {
    return BROWSER_WIDTH_RATIO.TABLET_9;
  }
  if (width > BROWSER_MIN_WIDTH.MOBILE) {
    return BROWSER_WIDTH_RATIO.TABLET_7;
  }
  return BROWSER_WIDTH_RATIO.MOBILE;
};

export default calcBrowserWidthRatio;
