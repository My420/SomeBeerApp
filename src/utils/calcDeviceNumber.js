import { DEVICE_MIN_WIDTH, DEVICE_NUMBER } from './constants';

const calcDeviceNumber = width => {
  if (width > DEVICE_MIN_WIDTH.LPC) {
    return DEVICE_NUMBER.LPC;
  }
  if (width > DEVICE_MIN_WIDTH.MPC) {
    return DEVICE_NUMBER.MPC;
  }
  if (width > DEVICE_MIN_WIDTH.TABLET) {
    return DEVICE_NUMBER.TABLET;
  }
  return DEVICE_NUMBER.MOBILE;
};

export default calcDeviceNumber;
