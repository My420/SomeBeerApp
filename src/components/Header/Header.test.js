import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import { NAVIGATION_LINKS } from '../../utils/constants';

describe('Test <Header /> component', () => {
  const props = {
    isOpen: true,
    isButtonShown: false,
    toggleState: () => {}
  };

  describe('should correct render component with PC props', () => {
    const wrapper = shallow(<Header {...props} />);

    test('should correct render <Header /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  render <header> tag with classNAme = header', () => {
      expect(wrapper.find('header.header')).toHaveLength(1);
    });

    test('should  render <Container> component with classNAme = inner', () => {
      expect(wrapper.find('Container.inner')).toHaveLength(1);
    });

    test('should  render <Navigation /> component ', () => {
      expect(wrapper.find('Navigation')).toHaveLength(1);
    });

    test('<Navigation /> component should have correct props links', () => {
      expect(wrapper.find('Navigation').prop('links')).toEqual(
        NAVIGATION_LINKS
      );
    });

    test('should not  render <Button/> component with className = button', () => {
      expect(wrapper.find('Button.button')).toHaveLength(0);
    });
  });

  describe('should correct render component with Tablet props (nav close)', () => {
    const newProps = { ...props, isOpen: false, isButtonShown: true };
    const wrapper = shallow(<Header {...newProps} />);

    test('should correct render <Header /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  render <header> tag with classNAme = header', () => {
      expect(wrapper.find('header.header')).toHaveLength(1);
    });

    test('should  render <Container> component with classNAme = inner', () => {
      expect(wrapper.find('Container.inner')).toHaveLength(1);
    });

    test('should not  render <Navigation /> component ', () => {
      expect(wrapper.find('Navigation')).toHaveLength(0);
    });

    test('should render <Button/> component with className = button ', () => {
      expect(wrapper.find('Button.button')).toHaveLength(1);
    });

    test('shuold render IconMenuCss inside <Button/> component ', () => {
      expect(wrapper.find('Button').find('IconMenuCss')).toHaveLength(1);
    });
  });

  describe('should correct render component with Tablet props (nav open)', () => {
    const newProps = { ...props, isOpen: true, isButtonShown: true };
    const wrapper = shallow(<Header {...newProps} />);

    test('should correct render <Header /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  render <header> tag with classNAme = header', () => {
      expect(wrapper.find('header.header')).toHaveLength(1);
    });

    test('should  render <Container> component with classNAme = inner', () => {
      expect(wrapper.find('Container.inner')).toHaveLength(1);
    });

    test('should render <Navigation /> component ', () => {
      expect(wrapper.find('Navigation')).toHaveLength(1);
    });

    test('<Navigation /> component should have correct props links', () => {
      expect(wrapper.find('Navigation').prop('links')).toEqual(
        NAVIGATION_LINKS
      );
    });

    test('should render <Button/> component with className = button ', () => {
      expect(wrapper.find('Button.button')).toHaveLength(1);
    });

    test('shuold render IconCrossCss inside <Button/> component ', () => {
      expect(wrapper.find('Button').find('IconCrossCss')).toHaveLength(1);
    });
  });

  describe('click on <Button/> component  should call toggleState function ', () => {
    test('should correct work when menu open', () => {
      const toggleFunc = jest.fn();
      const newProps = {
        ...props,
        isOpen: true,
        isButtonShown: true,
        toggleState: toggleFunc
      };
      const wrapper = shallow(<Header {...newProps} />);

      wrapper.find('Button').simulate('click');
      expect(toggleFunc.mock.calls.length).toBe(1);
    });

    test('should correct work when menu close', () => {
      const toggleFunc = jest.fn();
      const newProps = {
        ...props,
        isOpen: false,
        isButtonShown: true,
        toggleState: toggleFunc
      };
      const wrapper = shallow(<Header {...newProps} />);

      wrapper.find('Button').simulate('click');
      expect(toggleFunc.mock.calls.length).toBe(1);
    });
  });
});
