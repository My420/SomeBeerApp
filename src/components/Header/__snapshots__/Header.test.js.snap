// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Header /> component should correct render component with PC props should correct render <Header /> component  1`] = `
<header
  className="header"
>
  <Container
    className="inner"
  >
    <TransitionGroup
      childFactory={[Function]}
      component={null}
    >
      <CSSTransition
        classNames={Object {}}
        timeout={300}
      >
        <div>
          <Navigation
            links={
              Array [
                Array [
                  "/",
                  "Home",
                ],
                Array [
                  "/catalog",
                  "Catalog",
                ],
                Array [
                  "/roulette",
                  "Roulette!",
                ],
                Array [
                  "/favorite",
                  "Favorite",
                ],
                Array [
                  "/about",
                  "About",
                ],
                Array [
                  "/cart",
                  "Cart",
                ],
              ]
            }
          />
        </div>
      </CSSTransition>
    </TransitionGroup>
  </Container>
</header>
`;

exports[`Test <Header /> component should correct render component with Tablet props (nav close) should correct render <Header /> component  1`] = `
<header
  className="header"
>
  <Container
    className="inner"
  >
    <TransitionGroup
      childFactory={[Function]}
      component={null}
    />
  </Container>
  <Button
    activeStyle=""
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <IconMenuCss />
  </Button>
</header>
`;

exports[`Test <Header /> component should correct render component with Tablet props (nav open) should correct render <Header /> component  1`] = `
<header
  className="header"
>
  <Container
    className="inner"
  >
    <TransitionGroup
      childFactory={[Function]}
      component={null}
    >
      <CSSTransition
        classNames={Object {}}
        timeout={300}
      >
        <div>
          <Navigation
            links={
              Array [
                Array [
                  "/",
                  "Home",
                ],
                Array [
                  "/catalog",
                  "Catalog",
                ],
                Array [
                  "/roulette",
                  "Roulette!",
                ],
                Array [
                  "/favorite",
                  "Favorite",
                ],
                Array [
                  "/about",
                  "About",
                ],
                Array [
                  "/cart",
                  "Cart",
                ],
              ]
            }
          />
        </div>
      </CSSTransition>
    </TransitionGroup>
  </Container>
  <Button
    activeStyle=""
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <IconCrossCss />
  </Button>
</header>
`;
